packages/checkout-storefront/src/sections/AddressEditForm/AddressEditForm.tsx

```diff
@@ -0,0 +1,80 @@
+import { AddressFormData } from "@/checkout-storefront/components/AddressForm/types";
+import React from "react";
+import { AddressForm, AddressFormProps } from "@/checkout-storefront/components/AddressForm";
+import {
+  AddressFragment,
+  useUserAddressDeleteMutation,
+  useUserAddressUpdateMutation,
+} from "@/checkout-storefront/graphql";
+import { FormProvider } from "@/checkout-storefront/providers/FormProvider";
+import { useFormattedMessages } from "@/checkout-storefront/hooks/useFormattedMessages";
+import {
+  getAddressFormDataFromAddress,
+  getAddressInputData,
+} from "@/checkout-storefront/components/AddressForm/utils";
+import { useForm } from "@/checkout-storefront/hooks/useForm";
+import { useFormSubmit } from "@/checkout-storefront/hooks/useFormSubmit";
+import { AddressFormActions } from "@/checkout-storefront/components/ManualSaveAddressForm";
+import { addressEditMessages } from "@/checkout-storefront/sections/AddressEditForm/messages";
+import { useAddressFormSchema } from "@/checkout-storefront/components/AddressForm/useAddressFormSchema";
+import { useSubmit } from "@/checkout-storefront/hooks/useSubmit";
+
+export interface AddressEditFormProps extends Pick<AddressFormProps, "title"> {
+  address: AddressFragment;
+  onUpdate: (address: AddressFragment) => void;
+  onDelete: (id: string) => void;
+  onClose: () => void;
+}
+
+export const AddressEditForm: React.FC<AddressEditFormProps> = ({
+  onUpdate,
+  onClose,
+  onDelete,
+  address,
+}) => {
+  const formatMessage = useFormattedMessages();
+  const validationSchema = useAddressFormSchema();
+  const [{ fetching: updating }, userAddressUpdate] = useUserAddressUpdateMutation();
+  const [{ fetching: deleting }, userAddressDelete] = useUserAddressDeleteMutation();
+
+  const onSubmit = useFormSubmit<AddressFormData, typeof userAddressUpdate>({
+    scope: "userAddressUpdate",
+    onSubmit: userAddressUpdate,
+    parse: (formData) => ({ id: address.id, address: { ...getAddressInputData(formData) } }),
+    onSuccess: ({ result }) => {
+      onUpdate(result.data?.accountAddressUpdate?.address as AddressFragment);
```

Comment:
technically it shouldn't since it's a success handler, but the mutation return type doesn't reflect that

Reply:
So what if result.data?.accountAddressUpdate?.address is null or undefined?

Reply:
Confirmed with @korycins, if the api doesn't return errors it'll always return an address, otherwise it'd be a bug in core. As you can see in the code the onSuccess only happens if no errors are found so it's safe to assume here the address will exist

Reply:
So let's add a runtime assertion here instead of a type assertion.
```suggestion
      invariant(result.data?.accountAddressUpdate?.address, `some error message…`);
      onUpdate(result.data.accountAddressUpdate.address);
```